internal/bminventory/inventory.go

```diff
@@ -2987,6 +2987,31 @@ func (b *bareMetalInventory) processDiskSpeedCheckResponse(ctx context.Context,
 	return b.hostApi.SetDiskSpeed(ctx, h, diskPerfCheckResponse.Path, diskPerfCheckResponse.IoSyncDuration, exitCode, nil)
 }
 
+func (b *bareMetalInventory) updateDomainNameResolutionResponse(ctx context.Context, host *models.Host, domainResolutionResponseReporter string) error {
+	var (
+		err                      error
+		domainResolutionResponse models.DomainResolutionResponse
+	)
+	log := logutil.FromContext(ctx, b.log)
+	log.Debugf("The address for domain name resolotion on host %s are: %s", host.ID.String(), domainResolutionResponse)
+
+	if err = json.Unmarshal([]byte(domainResolutionResponseReporter), &domainResolutionResponse); err != nil {
+		log.WithError(err).Warnf("Json unmarshal domain name resolution of host %s", host.ID.String())
+		return err
+	}
+	if len(domainResolutionResponse.Resolutions) == 0 {
```

Comment:
Can we go a bit further here and make sure there is a resolution corresponding to each domain in the original request?

Reply:
I end up adding all the checks into the validation including this one, as Ori mention in one of the next comments it makes more sense to check the response over there.